spec/integration/slices_spec.rb

```diff
@@ -81,6 +81,188 @@ class Slice < Hanami::Slice
     end
   end
 
+  it "Loading a slice with its own defined slice class" do
+    with_tmp_directory(Dir.mktmpdir) do
+      write "config/app.rb", <<~RUBY
+        require "hanami"
+
+        module TestApp
+          class App < Hanami::App
+          end
+        end
+      RUBY
+
+      write "slices/main/config/slice.rb", <<~RUBY
+        module Main
+          class Slice < Hanami::Slice
+            config.actions.format :json
+          end
+        end
+      RUBY
+
+      require "hanami/prepare"
+
+      expect(Hanami.app.slices[:main]).to be Main::Slice
+      expect(Hanami.app.slices[:main].config.actions.format).to eq([:json])
+    end
+  end
+
+  it "Loading a slice with its own defined slice class prefers the app's defined slice class" do
+    with_tmp_directory(Dir.mktmpdir) do
+      write "config/app.rb", <<~RUBY
+        require "hanami"
+
+        module TestApp
+          class App < Hanami::App
+          end
+        end
+      RUBY
+
+      write "config/slices/main.rb", <<~RUBY
+        require "hanami"
+
+        module Main
+          class Slice < Hanami::Slice
+            config.actions.format :html
+          end
+        end
+      RUBY
+
+      write "slices/main/config/slice.rb", <<~RUBY
+        module Main
+          class Slice < Hanami::Slice
+            config.actions.format :json
+          end
+        end
+      RUBY
+
+      require "hanami/prepare"
+
+      expect(Hanami.app.slices[:main]).to be Main::Slice
+      expect(Hanami.app.slices[:main].config.actions.format).to eq([:html])
+    end
+  end
+
+  it "Loading a nested slice with its own defined slice class" do
+    with_tmp_directory(Dir.mktmpdir) do
+      write "config/app.rb", <<~RUBY
+        require "hanami"
+
+        module TestApp
+          class App < Hanami::App
+          end
+        end
+      RUBY
+
+      write "slices/main/slices/nested/config/slice.rb", <<~RUBY
+        module Main
+          module Nested
+            class Slice < Hanami::Slice
+              config.actions.format :json
+            end
+          end
+        end
+      RUBY
+
+      require "hanami/prepare"
+
+      expect(Hanami.app.slices[:main]).to be Main::Slice
+      expect(Hanami.app.slices[:main].slices[:nested].config.actions.format).to eq([:json])
+    end
+  end
+
+  it "Loading a nested slice with its own defined slice class prefers the parent slice's defined slice class" do
```

Comment:
Getting real _“How many slices would a Hanami app slice if a Hanami app could slice slices?”_ vibes from this description 😆

Reply:
ALL OF THEM